### ISSUE TITLE
iOS build ignoring ExportOptions.plist

### ISSUE THREAD

**Issue by DeerSteak** [hey24sheep/azure-flutter-tasks#28]
I used XCode to archive my package and export an Ad-Hoc version, which created ExportOptions.plist. I copied that file to my ./ios directory in the flutter project and added the following to my pipeline yaml:

```
- task: Hey24sheep.flutter.flutter-build.FlutterBuild@0
  inputs:
    target: 'ipa'
    exportOptions: plist
    exportOptionsPlist: './ios/ExportOptions.plist'
```

My ExportOptions.plist file looks like this, with the teamID redacted:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>compileBitcode</key>
	<true/>
	<key>destination</key>
	<string>export</string>
	<key>method</key>
	<string>ad-hoc</string>
	<key>signingStyle</key>
	<string>automatic</string>
	<key>stripSwiftSymbols</key>
	<true/>
	<key>teamID</key>
	<string>(redacted)</string>
	<key>thinning</key>
	<string>&lt;none&gt;</string>
</dict>
</plist>
```

I've installed an Ad-Hoc provisioning profile and my .p12 file with my certificate. Running 

`flutter build ipa --export-options-plist=./ios/exportOptions.plist`

on my local machine works and generates the IPA signed with the right ad-hoc profile and the cert from my machine. 

Unfortuantely it seems this file is ignored. The important bit is at the end where it says it's trying to find a DEVELOPMENT profile, rather than the Ad-Hoc one. 

```
2021-11-15T15:38:03.2166260Z ##[section]Starting: FlutterBuild
2021-11-15T15:38:03.2179080Z ==============================================================================
2021-11-15T15:38:03.2179510Z Task         : Flutter Build Task
2021-11-15T15:38:03.2179870Z Description  : Build a Flutter application project.
2021-11-15T15:38:03.2180150Z Version      : 0.3.0
2021-11-15T15:38:03.2180400Z Author       : Hey24sheep
2021-11-15T15:38:03.2180850Z Help         : [More Information](https://github.com/hey24sheep/azure-flutter-tasks)
2021-11-15T15:38:03.2181300Z ==============================================================================
2021-11-15T15:38:03.4400350Z [command]/Users/runner/hostedtoolcache/Flutter/2.5.3-stable/macos/flutter/bin/flutter build ipa --export-options-plist=./ios/exportOptions.plist
2021-11-15T15:38:09.5078210Z Running "flutter pub get" in flutter_tools...                       3.8s
2021-11-15T15:38:23.4056060Z Running "flutter pub get" in s...                                  13.8s
2021-11-15T15:38:41.5315790Z Archiving com.cas.il-connect...
2021-11-15T15:38:42.2972730Z Automatically signing iOS for device deployment using specified development team in Xcode project: (redacted)
2021-11-15T15:38:52.6832630Z Running pod install...                                              5.3s
2021-11-15T15:39:00.7376480Z Running Xcode build...                                          
2021-11-15T15:39:00.7380760Z Xcode archive done.                                          8.0s
2021-11-15T15:39:00.7386430Z Failed to build iOS app
2021-11-15T15:39:00.7388700Z Error output from Xcode build:
2021-11-15T15:39:00.7389900Z ↳
2021-11-15T15:39:00.7391560Z     --- xcodebuild: WARNING: Using the first of multiple matching destinations:
2021-11-15T15:39:00.7393010Z     { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
2021-11-15T15:39:00.7394540Z     { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
2021-11-15T15:39:00.7396060Z     { platform:iOS Simulator, id:CA504BAF-14C7-4882-8717-17861EF2FAE0, OS:15.0, name:iPad (7th generation) }
2021-11-15T15:39:00.7397520Z     { platform:iOS Simulator, id:2B15421A-35EB-442A-BC83-2BFFB5A918FB, OS:15.0, name:iPad (8th generation) }
2021-11-15T15:39:00.7399020Z     { platform:iOS Simulator, id:7A50DBD2-DBAB-4668-8647-63DA3C73746D, OS:15.0, name:iPad (9th generation) }
2021-11-15T15:39:00.7400460Z     { platform:iOS Simulator, id:EE083219-74C0-45EA-95D4-BE69A0BECF34, OS:15.0, name:iPad Air (3rd generation) }
2021-11-15T15:39:00.7401950Z     { platform:iOS Simulator, id:61E74034-9F1C-4DE2-92B6-78E85712E172, OS:15.0, name:iPad Air (4th generation) }
2021-11-15T15:39:00.7403380Z     { platform:iOS Simulator, id:A4FAE159-8EAD-4B67-9C9E-BC092323D327, OS:15.0, name:iPad Pro (9.7-inch) }
2021-11-15T15:39:00.7404860Z     { platform:iOS Simulator, id:4AFC00A4-1E34-410C-AEED-64739C5AC09D, OS:15.0, name:iPad Pro (11-inch) (2nd generation) }
2021-11-15T15:39:00.7406350Z     { platform:iOS Simulator, id:81891748-1799-4438-9229-C4DD21F8313B, OS:15.0, name:iPad Pro (11-inch) (3rd generation) }
2021-11-15T15:39:00.7407830Z     { platform:iOS Simulator, id:BFE2B277-C73F-4C01-92BA-65E681BBF07C, OS:15.0, name:iPad Pro (12.9-inch) (4th generation) }
2021-11-15T15:39:00.7409350Z     { platform:iOS Simulator, id:05A1339C-B80D-4C57-9A22-774417508E59, OS:15.0, name:iPad Pro (12.9-inch) (5th generation) }
2021-11-15T15:39:00.7410890Z     { platform:iOS Simulator, id:A2CE1931-8C20-4318-81EF-AEBFB5E8BDA8, OS:15.0, name:iPad mini (6th generation) }
2021-11-15T15:39:00.7412310Z     { platform:iOS Simulator, id:816B76CB-2635-46BD-9051-8F0AEC0B6D4F, OS:15.0, name:iPhone 8 }
2021-11-15T15:39:00.7413670Z     { platform:iOS Simulator, id:94F71856-FC49-4F3E-906C-8DD5804F22DD, OS:15.0, name:iPhone 8 Plus }
2021-11-15T15:39:00.7415050Z     { platform:iOS Simulator, id:424812ED-A78C-47F5-88A3-AD5901E2453F, OS:15.0, name:iPhone 11 }
2021-11-15T15:39:00.7416440Z     { platform:iOS Simulator, id:C01131F9-506B-4347-9DCC-339D669A78AC, OS:15.0, name:iPhone 11 Pro }
2021-11-15T15:39:00.7417840Z     { platform:iOS Simulator, id:FD7E4265-50DA-427E-AC3C-E5DD8B060491, OS:15.0, name:iPhone 11 Pro Max }
2021-11-15T15:39:00.7419240Z     { platform:iOS Simulator, id:71B10A8D-AE1C-4258-9D14-7501BB8683AE, OS:15.0, name:iPhone 12 }
2021-11-15T15:39:00.7420620Z     { platform:iOS Simulator, id:07667609-E2C2-4C30-995B-496E3ED059B9, OS:15.0, name:iPhone 12 Pro }
2021-11-15T15:39:00.7422020Z     { platform:iOS Simulator, id:55876080-B5DD-4F1F-AC97-9EF95B429AAE, OS:15.0, name:iPhone 12 Pro Max }
2021-11-15T15:39:00.7424080Z     { platform:iOS Simulator, id:D32E0883-F25E-4F15-9D52-E99F9A351004, OS:15.0, name:iPhone 12 mini }
2021-11-15T15:39:00.7425460Z     { platform:iOS Simulator, id:55AF4882-1EFE-4A74-A335-5B5D449B3C14, OS:15.0, name:iPhone 13 }
2021-11-15T15:39:00.7426820Z     { platform:iOS Simulator, id:43CCA3DA-0CFF-42FC-84A5-BF89E12CE1DF, OS:15.0, name:iPhone 13 Pro }
2021-11-15T15:39:00.7428220Z     { platform:iOS Simulator, id:5CF88143-1177-432A-A9A6-C398EF2DEBC7, OS:15.0, name:iPhone 13 Pro Max }
2021-11-15T15:39:00.7429620Z     { platform:iOS Simulator, id:C67BBF8B-6303-4D78-A81F-64957AA9499E, OS:15.0, name:iPhone 13 mini }
2021-11-15T15:39:00.7431550Z     { platform:iOS Simulator, id:B8BBB460-79BF-4410-91BE-9899091F08C3, OS:15.0, name:iPhone SE (2nd generation) }
2021-11-15T15:39:00.7432990Z     { platform:iOS Simulator, id:BBDFD409-46E9-44BC-8DF6-935D3A9CFDC0, OS:15.0, name:iPod touch (7th generation) }
2021-11-15T15:39:00.7433800Z     ** ARCHIVE FAILED **
2021-11-15T15:39:00.7434170Z 
2021-11-15T15:39:00.7434350Z 
2021-11-15T15:39:00.7435060Z Xcode's output:
2021-11-15T15:39:00.7435690Z ↳
2021-11-15T15:39:00.7436290Z     note: Using new build system
2021-11-15T15:39:00.7436860Z     note: Planning
2021-11-15T15:39:00.7437420Z     note: Build preparation complete
2021-11-15T15:39:00.7438080Z     note: Building targets in parallel
2021-11-15T15:39:00.7439870Z     error: No profiles for '(redacted)' were found: Xcode couldn't find any iOS App Development provisioning profiles matching '(redacted)'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'Runner' from project 'Runner')
2021-11-15T15:39:00.7440890Z 
2021-11-15T15:39:00.7516800Z Encountered error while archiving for device.
2021-11-15T15:39:00.7768360Z ##[error]Error: The process '/Users/runner/hostedtoolcache/Flutter/2.5.3-stable/macos/flutter/bin/flutter' failed with exit code 1
2021-11-15T15:39:00.7786750Z ##[section]Finishing: FlutterBuild
```

Any help would be greatly appreciated.

**Comment by hey24sheep** [comment]
I don't think the bug is file not found, otherwise it would have thrown plist option not found or invalid. 

Solution 1 - You could make a runtime environment variable for this file and try to use that. Check [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml).

Solution 2 - Recheck if the file exists and provisional profiles exist.

**Comment by hey24sheep** [comment]
I know that export plist works because of [this](https://github.com/hey24sheep/azure-flutter-tasks/issues/7). So, it is an error somewhere in your setup or profiles.

**Comment by DeerSteak** [comment]
Thanks for looking into it. I've got another member of the team who's looked at it and he's stumped too. The files are there, at least. 

We're going to have flutter just build an unsigned iOS .xcarchive and we'll do the distribution part afterwards. That should fix it.

**Comment by hey24sheep** [comment]
I see, you could try changing the path. You could check [this](https://github.com/hey24sheep/azure-flutter-tasks/issues/7). I am sorry, I am not of much help here as that argument is working on others. 

I am closing this issue as you have found a workaround. If you face any issue, feel free to ask for reopen or a new issue.
